src/core/identifiers/identifiers.test.ts

```diff
@@ -1,20 +1,56 @@
-import { buildDSNPAnnouncementURI, isDSNPAnnouncementURI, parseDSNPAnnouncementURI } from "./identifiers";
+import {
+  buildDSNPAnnouncementURI,
+  convertBigNumberToDSNPUserId,
+  convertBigNumberToDSNPUserURI,
+  convertDSNPUserIdOrURIToBigNumber,
+  convertDSNPUserURIToDSNPUserId,
+  isDSNPAnnouncementURI,
+  isDSNPUserURI,
+  parseDSNPAnnouncementURI,
+} from "./identifiers";
+import { BigNumber } from "ethers";
 
 describe("identifiers", () => {
+  describe("isDSNPUserURI", () => {
+    const validDSNPUserURIs = [
+      "dsnp://0xa123456789abcdef", // Lowercase
+    ];
+
+    const invalidDSNPUserURIs = [
+      "dsnp://0xa123456789ABCDEF", // Uppercase
+      "dsnp://0x0000456789abcdef", // Leading Zeros
+      "dsnp://0123456789abcdef", // No 0x user
+      "dsnp://0xbadwolf", // Bad user URI
+      "dsnp://0xa123456789abcdefa", // URI too long
+    ];
+
+    for (const id of validDSNPUserURIs) {
+      it(`returns true for "${id}"`, () => {
+        expect(isDSNPUserURI(id)).toEqual(true);
+      });
+    }
+
+    for (const id of invalidDSNPUserURIs) {
+      it(`returns false for "${id}"`, () => {
+        expect(isDSNPUserURI(id)).toEqual(false);
+      });
+    }
+  });
+
   describe("isDSNPAnnouncementURI", () => {
     const validDSNPAnnouncementURIs = [
-      "dsnp://0x0123456789ABCDEF/0x0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF", // Uppercase
-      "dsnp://0x0123456789abcdef/0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", // Lowercase
+      "dsnp://0xa123456789abcdef/0xa123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", // Lowercase
     ];
 
     const invalidDSNPAnnouncementURIs = [
-      "dsnp://0123456789ABCDEF/0x0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF", // No 0x on user
-      "dsnp://0x0123456789abcdef/0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", // No 0x on announcement
+      "dsnp://0xa123456789ABCDEF/0xa123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF", // Uppercase
+      "dsnp://0x0000456789abcdef/0x0000456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", // Leading Zeros
+      "dsnp://0123456789abcdef/0xa123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", // No 0x on user
+      "dsnp://0xa123456789abcdef/0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", // No 0x on announcement
       "dsnp://0123456789abcdef/0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef", // No 0x on either
-      "dsnp://0xbadwolf/0x0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF", // Bad user URI
-      "dsnp://0x0123456789ABCDEF/0xbadbadbad", // Bad message id
```

Comment:
why did we drop the bad message id test?

Reply:
`0xbadbadbad` is valid now.

Reply:
could we add one with some non-hex chars then? maybe `dsnp://0x1234567890/0xNotHexadecimal`?

Reply:
👍 new Line 52